### PR TITLE
Allow auth header of type AccessToken

### PR DESCRIPTION
**Pull request by rawsyntax**

### CODE REVIEW

**Comment by rawsyntax** [comment]
@igoihman @vkareh PTAL

**Comment by vkareh** [comment]
I'm not sure I understand the logic here (I'm not very familiar with the handler interface, to be honest), but shouldn't it call the `next` handler once it confirms that it has the `AccessToken`? So instead of returning, you can just skip to the end of the function (bypassing the bearer/claims check)

File: authentication/handler_test.go [hunk 1]
```diff
@@ -553,6 +553,31 @@ var _ = Describe("Handler", func() {
 		}`))
 	})
 
+	It("Allows access token authorization", func() {
+		// Prepare the next handler, which should not be called:
+		next := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			Expect(true).To(BeFalse())
```
Comment (vkareh) [inline authentication/handler_test.go:559]: This test will fail now, as the handler is continuing on to `next`

Comment (rawsyntax) [reply authentication/handler_test.go:559]: Fixing up. I was having some issues running the tests locally (passing?)

File: authentication/handler_test.go [hunk 2]
```diff
@@ -553,6 +553,30 @@ var _ = Describe("Handler", func() {
 		}`))
 	})
 
+	It("Allows access token authorization", func() {
+		// Prepare the next handler, which should not be called:
```
Comment (vkareh) [inline authentication/handler_test.go:557]: `s/should not/should/` :stuck_out_tongue:

Comment (rawsyntax) [reply authentication/handler_test.go]: thanks :)

**Comment by rawsyntax** [comment]
@vkareh fixed up

**Comment by vkareh** [comment]
Code /lgtm

Now how do I test it? :)

**Comment by rawsyntax** [comment]
Usage is in a separate repo I've sent you the link

**Comment by vkareh** [comment]
Thanks! /lgtm

**Pull request merged**